spec/factories/lead_provider.rb

```diff
@@ -4,7 +4,6 @@
   factory :lead_provider do
     name  { "Lead Provider" }
     cohorts { Cohort.all }
-    cpd_lead_provider
```

Comment:
This was creating an extra record causing test to have false positive

Reply:
I'd rather use the higher level factory `create(:cpd_lead_provider, :with_lead_provider).lead_provider`, that avoids creating extra db records in both directions. But I'll ✂️ this.